lib/tracks.py

```diff
@@ -0,0 +1,111 @@
+# types/dev
+from typing import Optional
+import numpy.typing as npt
+import logging
+# libs
+import numpy as np
+# other parts of this project
+from lib.constants import *
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    format="%(levelname)s: %(asctime)s %(name)s: %(message)s")
+
+
+class Track():
+    """A base class for other types of track.
+    """
+
+    def __init__(self, data: npt.NDArray[DTYPE]) -> None:
+        """Initialize a new instance of Track.
+
+        Args:
+            data (npt.NDArray[DTYPE], optional): The audio data. Defaults to np.empty((0, CHANNELS), dtype=DTYPE).
+        """
+        self.data = data
+
+
+class RecordedTrack(Track):
+    """A naive track which is currently being recorded. 
+    Does not check whether the values it holds make sense (stop_rec_time can be before start_rec_time etc.). 
+    """
+
+    def __init__(self, data: npt.NDArray[DTYPE] = np.empty((0, CHANNELS), dtype=DTYPE)) -> None:
+        """Initialize an instance of recorded track.
+
+        Args:
+            data (npt.NDArray[DTYPE], optional): The audio data. Defaults to np.empty((0, CHANNELS), dtype=DTYPE).
+        """
+        super().__init__(data)
+        self.first_frame_time: Optional[int] = None
```

Comment:
take podtrzitka?

Reply:
Tady ne, zvenku na ně šahám jinde v programu. Musel bych naprogramovat settery pro všechny tři.

Reply:
Pokud na ne sahate zvenku, pak je to v poradku. Settery nejsou potreba.